content/en/docs/components/pipelines/sdk/build-pipeline.md

```diff
@@ -0,0 +1,465 @@
+<a class="github-link" href="https://github.com/kubeflow/website/blob/master/content/en/docs/pipelines/sdk/build-pipeline.ipynb">View source on GitHub</a>
+</div>
+
+
+
+A Kubeflow pipeline is a portable and scalable definition of a machine learning
+(ML) workflow. Each step in your ML workflow, such as preparing data or
+training a model, is an instance of a pipeline component. This document
+provides an overview of pipeline concepts and best practices, and instructions
+describing how to build an ML pipeline.
+
+## Before you begin
+
+1.  Run the following command to install the Kubeflow Pipelines SDK. If you run this command in a Jupyter
+    notebook, restart the kernel after installing the SDK. 
+
+
+```python
+$ pip install kfp --upgrade
+```
+
+2.  Import the `kfp` and `kfp.components` packages.
+
+
+```python
+import kfp
+import kfp.components as comp
+```
+
+3. Create an instance of the [`kfp.Client` class][kfp-client]. To find your
+   Kubeflow Pipelines cluster's hostname and URL scheme, open the Kubeflow
+   Pipelines user interface in your browser. The URL of the Kubeflow
+   Pipelines user interface is something like 
+   `https://my-cluster.my-organization.com/pipelines`. In this case, the
+   host name and URL scheme are `https://my-cluster.my-organization.com`. 
+
+[kfp-client]: https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.client.html#kfp.Client
+
+
+```python
+# If you run this command on a Jupyter notebook running on Kubeflow, you can
+# exclude the host parameter.
+# client = kfp.Client()
+client = kfp.Client(host='<your-kubeflow-pipelines-host-name>')
+```
+
+## Understanding pipelines
+
+A Kubeflow pipeline is a portable and scalable definition of an ML workflow,
+based on containers. A pipeline is composed of a set of input parameters and a
+list of the steps in this workflow. Each step in a pipeline is an instance of a
+component, which is represented as an instance of 
+[`ContainerOp`][container-op].
+
+You can use pipelines to:
+
+*   Orchestrate repeatable ML workflows.
+*   Accelerate experimentation by running a workflow with different sets of
+    hyperparameters.
+
+### Understanding pipeline components
+
+A pipeline component is a containerized application that performs one step in a
+pipeline's workflow. Pipeline components are defined in
+[component specifications][component-spec], which define the following:
+
+*   The component's interface, its inputs and outputs.
+*   The component's implementation, the container image and the command to
+    execute.
+*   The component's metadata, such as the name and description of the
+    component.
+
+You can build components by [defining a component specification for a
+containerized application][component-dev], or you can [use the Kubeflow
+Pipelines SDK to generate a component specification for a Python
+function][python-function-component]. You can also [reuse prebuilt components
```

Comment:
Do you know whether we should use "prebuilt" or "pre-built"? I'd use the former, which is what you've got, but I'm not 100% sure it's the correct way, as the GCloud Docs use both https://cloud.google.com/s/results?q=prebuilt

Reply:
The Google Developer Docs Style Guide recommends using the closed form of compound words and words with prefixes. So, it looks like our preference is correct.

https://developers.google.com/style/word-list#compounds

Reply:
Awesome, that's good to know! Thanks @joeliedtke 👍